hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -98,24 +101,26 @@
    */
   private long contentRangeStart;
 
-  public S3AInputStream(String bucket,
-      String key,
-      long contentLength,
-      AmazonS3 client,
-      FileSystem.Statistics stats,
-      S3AInstrumentation instrumentation,
-      long readahead,
-      S3AInputPolicy inputPolicy) {
-    Preconditions.checkArgument(StringUtils.isNotEmpty(bucket), "No Bucket");
-    Preconditions.checkArgument(StringUtils.isNotEmpty(key), "No Key");
-    Preconditions.checkArgument(contentLength >= 0 , "Negative content length");
-    this.bucket = bucket;
-    this.key = key;
+  public S3AInputStream(S3ObjectAttributes s3Attributes,
+                        long contentLength,
+                        AmazonS3 client,
+                        FileSystem.Statistics stats,
+                        S3AInstrumentation instrumentation,
+                        long readahead,
+                        S3AInputPolicy inputPolicy) {
+    Preconditions.checkArgument(StringUtils.isNotEmpty(s3Attributes.getBucket()), "No Bucket");
```

Comment:
I wouldn't adjust the indentation, because it amplifies the diff and makes merging harder. Lines 105-110 should be the same as the original.